### ISSUE TITLE
Excel export render no date

### ISSUE THREAD

**Issue by jonathangreco** [ThaDafinser/ZfcDatagrid#250]
Hi,

i've noticed an issue with the 1.2.0 version when i try to render a grid in Excel, when i declare a Date :

``` PHP
        $col = new Column\Select('dateInvoice');
        $col->setLabel('Date Facture');
        $col->setWidth(5);
        $col->setType($this->getDateColType());
        $grid->addColumn($col);

// WHERE getDateColType() is equal to
    public function getDateColType()
    {
        $colType = new Type\DateTime(
            'Y-m-d',
            \IntlDateFormatter::SHORT,
            \IntlDateFormatter::NONE,
            'fr_FR'
        );

        return $colType;
    }
```

No date is shown juste FALSE in the Excel.

No problem in CSV file, i think there is a bug in the excel export about dates

**Comment by ThaDafinser** [comment]
@jonathangreco i was changed with this commit 
https://github.com/ThaDafinser/ZfcDatagrid/commit/d9b331dbcec582c42c6cf555969dc167d226e851

current point
https://github.com/ThaDafinser/ZfcDatagrid/blob/master/src/ZfcDatagrid/Renderer/PHPExcel/Renderer.php#L105-L114

is your input a DateTime object or is it a string?

**Comment by jonathangreco** [comment]
It's a string.

**Comment by jonathangreco** [comment]
Ok, I make some changes :

````PHP
case 'ZfcDatagrid\Column\Type\DateTime':
                        if ($value instanceof \DateTime) {
                            $value->setTimezone(new \DateTimeZone($col->getType()
                                ->getOutputTimezone()));
                        } else {
                            $value = new \DateTime($value);
                            $value->setTimezone(new \DateTimeZone($col->getType()
                                ->getOutputTimezone()));
                        }
                        $cell->setValue(\PHPExcel_Shared_Date::PHPToExcel($value));
                        $cell->getStyle()
                        ->getNumberFormat()
                        ->setFormatCode($col->getType()->getSourceDateTimeFormat());
                        break;
``````

**Comment by jonathangreco** [comment]
@ThaDafinser any feedback ?

**Comment by ThaDafinser** [comment]
@jonathangreco sry for the delay - i will have today in the evening some time (or tomorrow morning)

**Comment by ThaDafinser** [comment]
@jonathangreco i tried it with this successfully, can you try this snippet?

```php
                    case 'ZfcDatagrid\Column\Type\DateTime':
                        /* @var $dateType \ZfcDatagrid\Column\Type\DateTime */
                        $dateType = $col->getType();
                        
                        if (!$value instanceof \DateTime && !is_object($value)) {
                            $value = \DateTime::createFromFormat($dateType->getSourceDateTimeFormat(), $value);
                            $value->setTimezone(new \DateTimeZone($dateType->getSourceTimezone()));
                        }
                        
                        $value->setTimezone(new \DateTimeZone($dateType->getOutputTimezone()));
                        
                        $cell->setValue(\PHPExcel_Shared_Date::PHPToExcel($value));
                        $cell->getStyle()
                        ->getNumberFormat()
                        ->setFormatCode(\PHPExcel_Style_NumberFormat::FORMAT_DATE_DATETIME);
                        break;
```

**Comment by jonathangreco** [comment]
I try and give you feedback today.

**Comment by jonathangreco** [comment]
I've tested your snipper works for the export part, one only things my code do than yours is the output format in Excel :

Mine allow the exact output the developper wants in the excel. Well i've to define the output via this method

````PHP
 public function getDateColType()
    {
        $colType = new Type\DateTime(
            'Y-m-d',
            \IntlDateFormatter::SHORT,
            \IntlDateFormatter::NONE,
            'fr_FR'
        );

        $colType->setOutputDateType(\IntlDateFormatter::SHORT);
        $colType->setOutputPattern('dd/MM/yyyy');
        $colType->setOutputTimezone('GMT');

        return $colType;
    }
``````

But with yours i got in the Excel file :
25/7/16  15:30

My desired output was 'dd/MM/yyyy'. Like the code above. Can we fix this ?

**Comment by jonathangreco** [comment]
@ThaDafinser up :)

**Comment by ThaDafinser** [comment]
I just checked the PHPExcel part and your code should work - it has a converter builtin for PHP date format.

I will create a PR

**Comment by ThaDafinser** [comment]
@jonathangreco pls review https://github.com/ThaDafinser/ZfcDatagrid/pull/255

**Comment by jonathangreco** [comment]
@ThaDafinser  It should work, so i'm ok with that.

**Comment by jonathangreco** [comment]
I think about the documentation, we should now be more specific about the option of the colType Date and the output pattern don't you think ? from this link :
https://github.com/ThaDafinser/ZfcDatagrid/blob/master/docs/03.%20Columns.md#datetime

We should add (in a proper english, sorry i'm not native speaker):
If you want a specifc output format for your date, you should set`for e.g 
```PHP 
$colType->setOutputPattern('mm-dd-yyyy');```

**Comment by jonathangreco** [comment]
This issue is fixed by #255.
